main/src/library/BPlusTree.flix

```diff
@@ -349,6 +350,36 @@ mod BPlusTree {
         let (leaf, stamp) = BPlusTree.findLeaf(k, t);
         Node.getWithDefault(k, stamp, d, leaf, t)
 
+    ///
+    /// Applies `f` to all mappings `k => v` in `t` in parallel.
+    ///
+    /// This method unsafely removes the `r` effect when spawning threads.
+    ///
+    /// Not thread-safe.
+    ///
+    @Internal
+    @Parallel
+    pub def parForEach(f: k -> v -> Unit \ r, t: BPlusTree[k, v, r]): Unit \ r with Order[k] = region rc {
+        let threadNum = if(Node.computeHeight(t->root) >= 2) {
```

Comment:
Could you add a short comment about the `threadNum` number? im not sure what this ITE ensures

Also, for efficiency, it seems potentially wasteful to compute the full height only to check that it is `>= 2`. I guess they're not tall in practice.

Reply:
Oh I see that computeHeight is `O(Height)`